pkg/controller/replication_controller.go

```diff
@@ -220,6 +224,10 @@ func (rm *ReplicationManager) updatePod(old, cur interface{}) {
 	}
 	// TODO: Write a unittest for this case
 	curPod := cur.(*api.Pod)
+	if curPod.DeletionTimestamp != nil {
+		rm.deletePod(curPod)
```

Comment:
Same comment as above. Also, does this presume that the pod has already been "counted" by the ReplicationManager?